libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -172,6 +172,19 @@ private List<VcStoreItem> getSuccessfulVCStoreItems(List<VcStoreItem> vcStoreIte
         return successfulVCStoreItems;
     }
 
+    public Optional<Boolean> getVCSuccessStatus(String userId, String criIssuer)
+            throws ParseException {
+        List<VcStoreItem> vcStoreItems = getVcStoreItems(userId);
+        for (VcStoreItem vcStoreItem : vcStoreItems) {
+            SignedJWT vc = SignedJWT.parse(vcStoreItem.getCredential());
+            if (vc.getJWTClaimsSet().getIssuer().equals(criIssuer)) {
+                return Optional.of(
```

Comment:
Do we need to return an optional? The only place we use this we just call `get()` on the optional, so we should probably just return the boolean.

Reply:
If we cannot find any VC Itesm, we return Optional.empty() with isPresent() in the EvaluateGpg45ScoresHandler. Returning a boolean here would break this structure and unit tests failed. I think, it's better for it to remain as 'Optional' .